Ficek_QuantiApp/Ficek_QuantiApp/Domain/Tests/RocketClientTests/RocketClient+sadTests.swift

```diff
@@ -8,139 +8,221 @@ import RequestBuilder
 @testable import Networking
 @testable import RocketClient
 
-
 final class RocketClientSadTests: XCTestCase {
+  var cancellables = Set<AnyCancellable>()
+  
+  //    override func invokeTest() {
+  //        withDependencies {
+  //            $0.heightConverter = .live()
+  //            $0.enginesConverter = .live()
+  //            $0.secondStageConverter = .live()
+  //            $0.firstStageConverter = .live()
+  //            $0.massConverter = .live()
+  //            $0.diameterConverter = .live()
+  //            $0.rocketConverter = .live()
+  //            $0.rocketsConverter = .live()
+  //        } operation: {
+  //          super.invokeTest()
+  //        }
+  //      }
+  
+  func test_network_not_functioning_combine() throws {
+    let expectation = expectation(description: "Awaiting Success")
     var cancellables = Set<AnyCancellable>()
+    var errorRecieved = false
+    let urlResponse = URLResponse(
+      url: URL(string:"https://api.spacexdata.com/v4/rockets")!,
+      mimeType: nil,
+      expectedContentLength: 0,
+      textEncodingName: nil
+    )
     
-    override func invokeTest() {
-        withDependencies {
-            $0.heightConverter = .live()
-            $0.enginesConverter = .live()
-            $0.secondStageConverter = .live()
-            $0.firstStageConverter = .live()
-            $0.massConverter = .live()
-            $0.diameterConverter = .live()
-            $0.rocketConverter = .live()
-            $0.rocketsConverter = .live()
-        } operation: {
-          super.invokeTest()
-        }
+    let mockUrlRequester = URLRequester { _ in
+      Just((Data(), urlResponse))
+        .setFailureType(to: URLError.self)
+        .eraseToAnyPublisher()
+    }
+    
+    let failedNetworkClient = NetworkClient(
+      urlRequester: mockUrlRequester,
+      networkMonitorClient: .live(onQueue: .main)
+    )
+    
+    let sut = withDependencies { dependency in
+      dependency.networkClient = failedNetworkClient
+    } operation: {
+      RocketClient.liveValue
+    }
+    
+    let result = sut.fetchAllRocketsCombine()
+    
+    result
+      .sink(
+        receiveCompletion: { completion in
+          switch completion {
+          case .finished: XCTFail("Test should fail")
+          case let .failure(error):
+            if error.underlyingError is NetworkError {
+              errorRecieved = true
+              expectation.fulfill()
+            } else {
+              XCTFail("Error should be networkError")
+            }
+          }
+          
+        }, receiveValue: { _ in }
+      )
+      .store(in: &cancellables)
+    
+    waitForExpectations(timeout: 10)
```

Comment:
Nikdy nedavej ten timeout takhle vysokej!! Pokud ten test bude nekdy failovat, tak se to tady na 10s zasekne. Za me je prijatelny tak 0.1 max!